nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateRecord.java

```diff
@@ -0,0 +1,649 @@
+                    fakerDatatypeValueList.add(new AllowableValue(allowableValueName, allowableValueDisplayName, allowableValueDisplayName));
+                }
+            } catch (Exception e) {
+                // Ignore, this should indicate a Faker method that we're not interested in
+            }
+        }
+
+        // Add types manually for those Faker methods that generate data rather than getting it from a resource file
+        fakerDatatypeValueList.add(FT_FUTURE_DATE);
+        fakerDatatypeValueList.add(FT_PAST_DATE);
+        fakerDatatypeValueList.add(FT_BIRTHDAY);
+        fakerDatatypeValueList.add(FT_NUMBER);
+        fakerDatatypeValueList.add(FT_MD5);
+        fakerDatatypeValueList.add(FT_SHA1);
+        fakerDatatypeValueList.add(FT_SHA256);
+        fakerDatatypeValueList.add(FT_SHA512);
+        fakerDatatypeValues = fakerDatatypeValueList.toArray(new AllowableValue[]{});
+    }
+
+    static final PropertyDescriptor SCHEMA_TEXT = new PropertyDescriptor.Builder()
+            .name("generate-record-schema-text")
+            .displayName("Schema Text")
+            .description("The text of an Avro-formatted Schema used to generate record data. If this property is set, any user-defined properties are ignored.")
+            .addValidator(new AvroSchemaValidator())
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .required(false)
+            .build();
+    static final PropertyDescriptor RECORD_WRITER = new PropertyDescriptor.Builder()
+            .name("generate-record-record-writer")
+            .displayName("Record Writer")
+            .description("Specifies the Controller Service to use for writing out the records")
+            .identifiesControllerService(RecordSetWriterFactory.class)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor NUM_RECORDS = new PropertyDescriptor.Builder()
+            .name("generate-record--num-records")
+            .displayName("Number of Records")
+            .description("Specifies how many records will be generated for each outgoing FlowFile.")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .defaultValue("100")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor LOCALE = new PropertyDescriptor.Builder()
+            .name("generate-record-locale")
+            .displayName("Locale")
+            .description("The locale that will be used to generate field data. For example a Locale of 'es' will generate fields (e.g. names) in Spanish.")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .defaultValue("en-US")
+            .allowableValues(SUPPORTED_LOCALES)
+            .build();
```

Comment:
As noted in the Windows build failure, some methods in `Faker` classes do not honor the `Locale` setting and fall back to the platform default Locale. Does it make sense to remove this as a configurable property for now and simply hard-code the implementation to `en-US` for the purpose of more predictable results?

Reply:
That's unfortunate but if Faker is buggy that way then we don't want to run into errors as a result, will remove, we can add it back later if Faker is fixed to honor the Locale settings